source/unified-test-format/unified-test-format.rst

```diff
@@ -1743,14 +1775,19 @@ to block indefinitely. This should not be a concern for
 ``iterateUntilDocumentOrError`` as iteration only continues until either a
 document or error is encountered.
 
-Test runners MUST ensure that this operation will not inadvertently skip the
-first document in a cursor. Albeit rare, this could happen if
-``iterateUntilDocumentOrError`` were to blindly invoke ``next`` (or
-equivalent) on a cursor in a driver where newly created cursors are already
-positioned at their first element and the cursor had a non-empty
-``firstBatch`` (i.e. ``resumeAfter`` or ``startAfter`` used). Alternatively,
-some drivers may use a different iterator method for advancing a cursor to
-its first position (e.g. ``rewind`` in PHP).
+iterateOnce
+~~~~~~~~~~~
+
+Performs a single iteration of the cursor. If the cursor's current batch is
+empty, one ``getMore`` MUST be attempted to get more results. This operation
```

Comment:
>If the cursor's current batch is empty, one ``getMore`` MUST be attempted to get more results

so, this means that we will iterate until a document like `iterateUntilDocumentOrError`. Then the only difference that this step is not expected to fail?

Reply:
The difference is that even if the `getMore` returns an empty batch, the driver ends the operation with an empty result. This is useful for cases like change streams. IIRC, it's used in a change streams test to assert that a `getMore` was sent and the timeout was applied to the command but we don't care about the result.

Reply:
then we have a bug in `IterateUntilDocumentOrErrorOperation`: https://github.com/mongodb/mongo-csharp-driver/blob/master/tests/MongoDB.Driver.Tests/UnifiedTestOperations/UnifiedIterateUntilDocumentOrErrorOperation.cs#L38, the result can contain an empty batch, so we don't wait until the document appearance